[CBRD-25862] System catalog naming convention

kangmin5505 opened the pull request:
[<http://jira.cubrid.org/browse/CBRD-25862>
](http://jira.cubrid.org/browse/CBRD-25862)

### Purpose
현재 catalog table의 prefix rule이 system table은 _db 로 시작하고 system view는 db 로 시작하는 rule를 위반하는 것이 있음
따라서 naming convention을 유지하고자 함
- 예외 system table : db_authorization, ~db_authorizations~, db_ha_apply_info, db_password, db_root, db_serial, db_trigger, db_user
​

하지만 authorization classes(db_authorization, db_password, db_root, db_user)는 system catalog classes(db_ha_apply_info, db_serial, db_trigger)와 구분 지어 다루어지므로 이후 필요 시 분석 후 수정

system catalog class 이름 변경
- `db_ha_apply_info` -> `_db_ha_apply_info`
- `db_serial` -> `_db_serial`
- `db_trigger` -> `_db_trigger`

system catalog view 이름 변경 및 생성
- `db_ha_apply_info` 생성
- `db_serial` 생성
- `db_trig` -> `db_trigger`


### Remarks

1. `db_ha_apply_info`의 경우 ha 관리 용도의 테이블이므로 일반 사용자에게 보여질 필요가 없다고 생각하여 view를 생성하지 않음.
     => 필요하다는 의견이 있을 시 생성
2. `db_serial` system catalog view에 class method가 추가되었는데, 이는 호환성을 위한 용도로 class method 또는 instance method의 지원이 종료되면 제거
     => 추가적으로 `change_serial_owner`를 `db_root`로 옮기는 방법도 고려 (`change_trigger_owner`의 경우 `db_root`의 class method)

kangmin5505 committed:
refactor: db_serial
- rename system catalog class `db_serial` to `_db_serial` to follow system class naming conventions
- introduce a new system catalog view `db_serial` that exposes `_db_serial`

kangmin5505 committed:
refactor: db_ha_apply_info
- rename system catalog class `db_ha_apply_info` to `_db_ha_apply_info`
to follow system catalog class naming conventions

kangmin5505 committed:
refactor: db_trigger
- rename system catalog class `db_trigger` to `_db_trigger` to follow
system catalog class naming convention
- rename system catalog view `db_trig` to `db_trigger` to follow system
catalog view naming convention

kangmin5505 committed:
codestyle: rollback

kangmin5505 commented:
현재 `contrib/php4/src/php_cubrid.c`파일로 인해 ci에서 codestyle fail이 발생하고 있는데, 코드 리뷰가 완료된 후 codestyle을 적용하여 merge를 하도록 하겠습니다.

kangmin5505 commented:
1. `db_ha_apply_info의 경우 ha 관리 용도의 테이블이므로 일반 사용자에게 보여질 필요가 없다고 생각하여 view를 생성하지 않음.`

2. `추가적으로 change_serial_owner를 db_root로 옮기는 방법도 고려 (change_trigger_owner의 경우 db_root의 class method)`
    - 위와 같은 방법으로 처리할 경우 해당 PR에서 system catalog view에 class method를 사용할 수 있도록 추가한 로직을 제거하여 기존 코드의 일관성을 유지할 수 있습니다.

위에 대해서 의견을 주시면 감사하겠습니다.

hyunikn commented on src/parser/semantic_check.c:
```diff
@@ -10346,7 +10346,7 @@ pt_check_alter_serial (PARSER_CONTEXT * parser, PT_NODE * node)
   assert (node->node_type == PT_ALTER_SERIAL);
 
-  /* find db_serial_class */
+  /* find _db_serial_class */
```
```suggestion
  /* find _db_serial class */
```

hyunikn commented on src/query/execute_statement.c:
```diff
@@ -1446,7 +1446,7 @@ do_create_serial (PARSER_CONTEXT * parser, PT_NODE * statement)
 
   /*
-   * find db_serial_class
+   * find _db_serial_class
```
```suggestion
   * find _db_serial class
```

hyunikn commented on src/query/execute_statement.c:
```diff
@@ -2382,7 +2382,7 @@ do_alter_serial (PARSER_CONTEXT * parser, PT_NODE * statement)
 
   /*
-   * find db_serial_class
+   * find _db_serial_class
```
```suggestion
   * find _db_serial class
```

kangmin5505 committed:
refactor: remove checking authorization of db_serial.(db_serial doesn't have authorization)

kangmin5505 committed:
apply review: `_db_serial_class` to `_db_serial class`

shparkcubrid commented:
view로 변경시에 의도치 않게 plan이 변경될 가능성이 있습니다.
그리고 db_ha_apply_info는 뷰를 만들지 않으면 하위 호환성 문제가 발생할텐데요. 기술본부 혹은 큐브리드 사용자가 조회해 볼 수 있고, 모니터링 프로그램에서 사용될 수 있을 것 같습니다. 참고하세요.
제 소유의 코드는 주석만 수정이 되었기 때문에 저는 리뷰를 진행하지 않겠습니다.

youngjinj commented on src/object/schema_system_catalog_install.cpp:
```diff
@@ -1784,6 +1777,53 @@ namespace cubschema
+		   CTV_SERIAL_NAME,
+		   // columns
+    {
+      {"unique_name", format_varchar (255)},
```
시스템 뷰에서는 `unique_name`을 보여주지 않아야 합니다.

kangmin5505 replied:
감사합니다.
반영하겠습니다.

youngjinj commented on src/optimizer/query_graph.c:
```diff
@@ -7289,8 +7289,8 @@ qo_find_node_indexes (QO_ENV * env, QO_NODE * nodep)
 	{
-	  /* Do not use index of db_serial/db_has_apply_info for scanning. Current index scanning is optimized for
-	   * MVCC, while db_serial and db_ha_apply_info have MVCC disabled.
+	  /* Do not use index of _db_serial/db_has_apply_info for scanning. Current index scanning is optimized for
```
```suggestion
	  /* Do not use index of _db_serial/_db_ha_apply_info for scanning. Current index scanning is optimized for
```

kangmin5505 commented:
@shparkcubrid
리뷰 감사합니다. `db_ha_apply_info`와 관련하여 말씀해 주신 부분에 대해서는 확인 후 진행하도록 하겠습니다.

kangmin5505 committed:
refactor: apply review
- remove unique_name on db_serial view
- fix comment from `db_has_apply_info` to `_db_ha_apply_info`

mhoh3963 commented:
@kangmin5505 
db_ha_apply_info는 system table이기는 한데, HA 재구성 작업을 위해 사용자(기술지원, 운영자 등)가 csql --sysadm을 이용해서 데이터 조작 작업(update, delete)을 수행합니다. 검색은 대부분은 applyinfo utility를 통해 검색하는데, 가끔 직접 table을 검색하는 경우도 있습니다.
db_ha_apply_info를 _db_ha_apply_info 로 변경시 UX (user experience) 가 변경되어 이 부분은 PM팀과 논의가 필요해 보이네요. (view에서 update, delete가 되지 않기때문에 완벽하게 호환하기 힘들어 보입니다.)

kangmin5505 commented:
@mhoh3963 
리뷰 감사합니다.
`db_ha_apply_info` 수정은 PM팀과 논의 후 진행하도록 하겠습니다.

kangmin5505 committed:
fix: add CTV_SERIAL_NAME to prohibited_classes in unload_object.c

kangmin5505 committed:
feature: add system catalog view `db_ha_apply_info` for user-friendly DML access
- Although it provides the same output as the `_db_ha_apply_info` class, this view improves user experience by allowing consistent DML-style access and naming conventions.

kangmin5505 committed:
refactor: codestyle

kangmin5505 commented:
`db_ha_apply_info` 시스템 카탈로그 뷰 추가 - 339450addf885418ddffe89667217705e214d230
- 목적: 기존 사용자(기술지원, 운영자)들의 UX를 지원하기 위함(ex. DML)
- 참고: 해당 파일의 수정은 시스템 카탈로그 클래스는 DONT_REUSE_OID 이기 때문에 뷰를 DML 을 사용할 수 있어야 함에도 불구하고, 정상 작동하지 않은 부분
```c
// src/parser/view_transform.c
      for (i = 0; i < *num_classes; ++i)
	{
	  is_reuse_oid_class = sm_is_reuse_oid_class ((*classes)[i]);

	  if (is_reuse_oid_class
	      || (!BOOT_ADMIN_CSQL_CLIENT_TYPE (client_type) && (sm_is_system_class ((*classes)[i]) > 0)))
	    {
	      local = (PT_UPDATABILITY) (local & PT_NOT_UPDATABLE);
	      if (parser->view_cache)
		{
		  parser->view_cache->has_reuse_oid_table = is_reuse_oid_class;
		}
	      break;
	    }
	}

```
- sysadm 권한이 아닌 상태에서 뷰를 사용하여 시스템 카탈로그 클래스 DML 작업 시, 에러 메시지.
  - 기존
    - `ERROR: Vclass db_ha_apply_info is not updatable. Please check if any of its related classes are marked as REUSE_OID.` --> 시스템 클래스는 DONT_REUSE_OID 임에도 불구하고 REUSE_OID와 관련이 있다는 에러 메시지 출력
  - 변경 후
    - `ERROR: Vclass db_ha_apply_info is not updatable.`

kangmin5505 committed:
refactor: avoid redundant db_get_client_type() and reuse
sm_is_reuse_oid_class() result

kangmin5505 committed:
codestyle
